### PR TITLE
CodeCharts

### PR DESCRIPTION
**Pull request by justinexner**

### CODE REVIEW

File: src/main/java/github/weichware10/toolbox/codecharts/CodeChartsController.java [hunk 1]
```diff
@@ -28,9 +41,35 @@ void endTest() {
...
+        codeChartsImageView.setFitWidth(width.get());
+        codeChartsGridPane.setPrefWidth(width.get());
+    }
+
+    private void setImageViewHeight(ObservableDoubleValue height) {
+        codeChartsImageView.setFitHeight(height.get());
+        codeChartsGridPane.setPrefHeight(height.get());
+    }
 
+    private void setGridPane(){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/codecharts/CodeChartsController.java:54]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/codecharts/CodeChartsInputController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package github.weichware10.toolbox.codecharts;
+
+import java.net.URL;
+import java.util.ResourceBundle;
+import javafx.fxml.FXML;
+import javafx.scene.control.ProgressBar;
+import javafx.scene.control.TextField;
+import javafx.scene.text.Text;
+
+public class CodeChartsInputController {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/codecharts/CodeChartsInputController.java:10]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java [hunk 3]
```diff
@@ -1,81 +1,193 @@
...
+        rootPane = new CodeChartsPane(null, -1, -1, width, height);
+        rootPane.subdivide(horizontal, vertical);
+        rootPane.philLeaves(usableStrings, currentStrings);
+        rootPane.setVisible(false);
+        stackPane.getChildren().add(rootPane);
     }
 
     /**
-     * Gibt Fehlermeldung bei falscher Config data.
+     * blbla.
```
Comment (TheDarkDesync) [inline src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java:116]:
```suggestion
     * Durchführen einer CodeCharts Iteration
```

File: src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java [hunk 4]
```diff
@@ -1,81 +1,193 @@
...
+            protected Void call() throws Exception {
+                try {
+                    Platform.runLater(() -> imageView.setVisible(true));
+                    Thread.sleep(timings[0]);
+                    Platform.runLater(() -> imageView.setVisible(false));
+                    Platform.runLater(() -> rootPane.setVisible(true));
+                    Thread.sleep(timings[1]);
+                    Platform.runLater(() -> rootPane.setVisible(false));
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
```
Comment (TheDarkDesync) [inline src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java:133]: ist das gewollt, dass wir den StrackTrace nicht als Logger.Error ausgeben?

Comment (joshuajeschek) [reply src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java:133]: nein.

File: src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java [hunk 5]
```diff
@@ -1,81 +1,194 @@
...
+import javafx.concurrent.Task;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
 
 /**
- * Steuert den Datenverkehr und gibt die BEfehle zum anzeigen.
+ * Steuert den Datenverkehr und gibt die Befehle zum anzeigen.
```
Comment (grenoui11e) [inline src/main/java/github/weichware10/toolbox/codecharts/CodeChartsCoordinator.java:24]:
```suggestion
 * Steuert den Datenverkehr und gibt die Befehle zum Anzeigen.
```

**Review by grenoui11e — APPROVED** [review]

**Review by Rotwolf — APPROVED** [review]

**Pull request merged**